packages/app-headless-cms/src/admin/components/ContentEntryForm/Header/DeleteEntry/DeleteEntry.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useCallback } from "react";
+import { ReactComponent as DeleteIcon } from "@material-design-icons/svg/filled/delete.svg";
+import { useRouter } from "@webiny/react-router";
+
+import { usePermission } from "~/admin/hooks/usePermission";
+import { ContentEntryEditorConfig } from "~/admin/config/contentEntries";
+import { useContentEntry } from "~/admin/views/contentEntries/hooks/useContentEntry";
+import { useDeleteEntry } from "./useDeleteEntry";
+
+export const DeleteEntry: React.FC = () => {
+    const { history } = useRouter();
+    const { entry, contentModel, loading } = useContentEntry();
+    const { canDelete } = usePermission();
+    const { showConfirmationDialog } = useDeleteEntry();
+    const { OptionsMenuItem } =
+        ContentEntryEditorConfig.Actions.MenuItemAction.useOptionsMenuItem();
+
+    if (!canDelete(entry, "cms.contentEntry")) {
+        return null;
+    }
+
+    const navigateBacktoList = useCallback(
```

Comment:
You should never use the React hook after `if` which breaks the component execution. This will break the UI eventually.